lib/fetchContent.js

```diff
@@ -323,6 +324,141 @@ function getRecentArticles(days, aspects, facets)
 		});
 }
 
+function getArticleTopics(days, aspects, facets)
+{
+	const date = time.getDatetimeRange('days', days, 0);
+	const params = {
+		queryString: `lastPublishDateTime:> ${date.next}`,
+		maxResults : 100,
+		aspects : aspects,
+		facets : {
+			"names": facets,
+			"maxElements":-1
+		}
+	};
+
+	return searchDeeper(params, 10)
+		.then(results =>  extractArticles(results) )
+		.then(articles => sortArticlesIntoTopics(articles) )
+		.catch(err => {
+			throw err;
+		});
+}
+
+function sortArticlesIntoTopics(articles)
+{
+	let allTopics = [];
+	let uniqueTopics = [];
+	let topicsWithArticles = [];
+
+	// Get all topics covered by articles
+	articles.forEach(article => {
```

Comment:
Wondering if a map/reduce would be more efficient than a forEach here

Reply:
Will look into this..

Reply:
I think I need the forEach as I don't need a new article array just a list to iterate against